### PR TITLE
Feat/push/add sync

### PR DESCRIPTION
**Pull request by Elyniss**

### CODE REVIEW

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/domain/SyncEngine.kt [hunk 1]
```diff
@@ -75,7 +79,7 @@ internal class SyncEngine(
     }
 
     fun setup() {
-        scope.launch { registerTagsInHistory() }
+//        scope.launch { registerTagsInHistory() }
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/domain/SyncEngine.kt:82]: Let's remove this line

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/use_case/calls/GetStoreTopicUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import com.walletconnect.android.sync.storage.StoresStorageRepository
+import com.walletconnect.foundation.common.model.Topic
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.runBlocking
+
+internal class GetStoreTopicUseCase(private val storesRepository: StoresStorageRepository) : GetStoreTopicUseCaseInterface {
+
+    override fun getStoreTopic(accountId: AccountId, store: Store): Topic? {
```
Comment (TalhaAli00) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/use_case/calls/GetStoreTopicUseCase.kt:15]: Can this be a suspend fun?

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt [hunk 3]
```diff
@@ -130,7 +130,7 @@ internal class ChatEngine(
     }
 
     fun setup() {
-        scope.launch { registerTagsInHistory() }
+//        scope.launch { registerTagsInHistory() }
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt:133]: Remove

File: push/sdk/src/main/kotlin/com/walletconnect/push/common/data/storage/SubscribeStorageRepository.kt [hunk 4]
```diff
@@ -2,103 +2,112 @@ package com.walletconnect.push.common.data.storage
...
 import com.walletconnect.push.common.model.EngineDO
-import com.walletconnect.push.common.storage.data.dao.SubscriptionsQueries
+import com.walletconnect.push.common.storage.data.dao.ActiveSubscriptionsQueries
+import com.walletconnect.push.common.storage.data.dao.RequestedSubscriptionQueries
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
+import timber.log.Timber
 
-class SubscribeStorageRepository(private val subscriptionQueries: SubscriptionsQueries) {
+class SubscribeStorageRepository(
```
Comment (TalhaAli00) [inline push/sdk/src/main/kotlin/com/walletconnect/push/common/data/storage/SubscribeStorageRepository.kt:15]: Let's change this to `SubscriptionRepository`

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt [hunk 5]
```diff
@@ -121,6 +138,7 @@ internal class PushWalletEngine(
     }
 
     fun setup() {
+        scope.launch { registerTagsInHistory() }
```
Comment (TalhaAli00) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt:141]: Let's make this suspend and call launch in the protocol

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt [hunk 6]
```diff
@@ -527,7 +562,7 @@ internal class PushWalletEngine(
         try {
             jsonRpcInteractor.respondWithSuccess(request, irnParams)
             // TODO: refactor to use the RPC published at value 
-            val currentTime = CURRENT_TIME_IN_SECONDS
+            val currentTime = request.id
```
Comment (TalhaAli00) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt:565]: Was this change intentional?

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/use_case/requests/DeleteSubscriptionWithSymmetricKeyToPushSubscriptionStoreUseCase.kt [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import com.walletconnect.foundation.common.model.Topic
+import com.walletconnect.foundation.util.Logger
+import com.walletconnect.push.wallet.engine.sync.PushSyncStores
+
+internal class DeleteSubscriptionToPushSubscriptionStoreUseCase(
+    private val logger: Logger,
+    private val syncClient: SyncInterface,
+) {
+
+    operator fun invoke(accountId: AccountId, pushTopic: Topic, onSuccess: (Boolean) -> Unit, onError: (Throwable) -> Unit) {
```
Comment (TalhaAli00) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/use_case/requests/DeleteSubscriptionWithSymmetricKeyToPushSubscriptionStoreUseCase.kt:16]: Make suspend?

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/use_case/requests/SetSubscriptionWithSymmetricKeyToPushSubscriptionStoreUseCase.kt [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import com.walletconnect.push.wallet.engine.sync.model.toSync
+
+internal class SetSubscriptionWithSymmetricKeyToPushSubscriptionStoreUseCase(
+    private val logger: Logger,
+    private val syncClient: SyncInterface,
+    _moshi: Moshi.Builder,
+) {
+    private val moshi = _moshi.build()
+
+    operator fun invoke(subscription: EngineDO.Subscription.Active, symmetricKey: SymmetricKey, onSuccess: (Boolean) -> Unit, onError: (Throwable) -> Unit) {
```
Comment (TalhaAli00) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/use_case/requests/SetSubscriptionWithSymmetricKeyToPushSubscriptionStoreUseCase.kt:21]: Can you make this suspend?

**Review by TalhaAli00 — APPROVED** [review]

File: web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/sync/di/SyncProxyModule.kt [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.walletconnect.web3.inbox.sync.event.OnSyncUpdateEventUseCase
+import com.walletconnect.web3.inbox.sync.event.SyncEventHandler
+import org.koin.dsl.module
+
+@JvmSynthetic
+internal fun syncProxyModule(
+) = module {
+    single { OnSyncUpdateEventUseCase(get(), get()) }
+    single { SyncEventHandler(get(), get(), get()) }
```
Comment (TalhaAli00) [inline web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/sync/di/SyncProxyModule.kt:13]: let's add named arguments here

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt (hunk 6 above)
Comment (Elyniss) [reply push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt:565]: Yeah, after messages are fetched from sync they all will have the same time. Until we move to publishedAt it's better value

File: web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/ui/WebView.kt [hunk 10]
```diff
@@ -207,7 +208,7 @@ open class AccompanistWebViewClient : WebViewClient() {
 
     override fun doUpdateVisitedHistory(view: WebView?, url: String?, isReload: Boolean) {
         super.doUpdateVisitedHistory(view, url, isReload)
-
+        Timber.d(url)
```
Comment (Elyniss) [inline web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/ui/WebView.kt:211]: Remove

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/domain/SyncEngine.kt [hunk 11]
```diff
@@ -101,7 +105,9 @@ internal class SyncEngine(
 
 
     private suspend fun registerTagsInHistory() {
-        historyInterface.registerTags(tags = listOf(Tags.SYNC_SET, Tags.SYNC_DELETE), {}, {})
+        // Has to be one register call per clientId
+        // Currently moved to PushWalletEgnine
```
Comment (jakubuid) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/domain/SyncEngine.kt:108]: Let's leave TODO for a better findability

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/model/SyncedSubscription.kt [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+
+internal fun SyncedMetadata.toCommon() = AppMetaData(
+    name = name,
+    description = description,
+    url = url,
+    icons = icons,
+    redirect = redirect?.toCommon(),
+//    verifyUrl = null //todo: Missing in Metadata in Core data structure
```
Comment (jakubuid) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/model/SyncedSubscription.kt:117]: Screenshot 2023-07-11 at 13 09 34

Comment (jakubuid) [reply push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/model/SyncedSubscription.kt:117]: From Core.Model

File: samples/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/Web3WalletActivity.kt [hunk 13]
```diff
@@ -148,7 +149,8 @@ class Web3WalletActivity : ComponentActivity() {
                         }
 
                         withContext(Dispatchers.Main) {
-                            navController.navigate(route.toString())
+                            Timber.d("Received PushProposal but it's handled by Web3Inbox")
+//                            navController.navigate(route.toString())
```
Comment (jakubuid) [inline samples/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/Web3WalletActivity.kt:153]: Remove or uncomment?

File: web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/client/Web3Inbox.kt [hunk 14]
```diff
@@ -59,7 +58,7 @@ object Web3Inbox {
     }
 
     private fun onPageFinished() = wrapWithInitializationCheck {
-        chatEventHandler = wcKoinApp.koin.get()
+        // Is this still being used somewhere?
```
Comment (jakubuid) [inline web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/client/Web3Inbox.kt:61]: Let's remove if not

File: web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/client/Web3Inbox.kt [hunk 15]
```diff
@@ -75,6 +74,7 @@ object Web3Inbox {
         return block()
     }
 
+
```
Comment (jakubuid) [inline web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/client/Web3Inbox.kt:77]: lol

**Review by jakubuid — APPROVED** [review]

File: web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/client/Web3Inbox.kt (hunk 15 above)
Comment (Elyniss) [reply web3/inbox/src/main/kotlin/com/walletconnect/web3/inbox/client/Web3Inbox.kt:77]: lol

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/model/SyncedSubscription.kt (hunk 12 above)
Comment (Elyniss) [reply push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/sync/model/SyncedSubscription.kt:117]: But is not available at that time, we do not store verifyUrl for metadata

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/domain/SyncEngine.kt [hunk 16]
```diff
@@ -101,7 +104,9 @@ internal class SyncEngine(
...
```
Comment (Elyniss) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/sync/engine/domain/SyncEngine.kt:108]:
```suggestion
        // TODO: Has to be one register call per clientId
        //  Currently moved to PushWalletEgnine
```
